### ISSUE TITLE
Api request not filtering by observation category

### ISSUE THREAD

**Issue by androBermudez** [Aidbox/Issues#27]
Url api request: `/Observation/$lastn?_count=200&_totalMethod=count&subject=pat1&category=vital-signs&max=1`

The response includes both `laboratory` and `vital-sign` observations:
```
{
  "resourceType": "Bundle",
  "type": "searchset",
  "total": 10,
  "query-sql": [
    "WITH codes AS (SELECT distinct(jsonb_array_elements(\"observation\".\"resource\"#>'{code,coding}')) AS code FROM \"observation\" WHERE true) SELECT grp.* FROM codes JOIN LATERAL (SELECT * FROM observation WHERE (true AND resource @> jsonb_build_object('code', jsonb_build_object('coding', jsonb_build_array(jsonb_build_object('code', codes.code->>'code', 'system', codes.code->>'system'))))) ORDER BY knife_extract_max_timestamptz(observation.resource, '[[\"effective\", \"dateTime\"], [\"effective\", \"period\", \"end\"]]') DESC LIMIT ? OFFSET ?) GRP ON (true)",
    1,
    0
  ],
  "entry": [
    {
      "resource": {
        "category": [
          {
            "text": "Laboratory",
            "coding": [
              {
                "code": "laboratory",
                "system": "http://hl7.org/fhir/observation-category",
                "display": "Laboratory"
              }
            ]
          }
        ],
        "referenceRange": [
          {
            "high": {
              "code": "mmol/L",
              "unit": "mmol/L",
              "value": 3,
              "system": "http://unitsofmeasure.org"
            }
          }
        ],
        "meta": {
          "lastUpdated": "2018-11-29T20:07:30.592Z",
          "versionId": "334",
          "tag": [{ "system": "https://aidbox.app", "code": "updated" }]
        },
        "value": {
          "Quantity": {
            "code": "mmol/L",
            "unit": "mmol/L",
            "value": 4.6,
            "system": "http://unitsofmeasure.org"
          }
        },
        "resourceType": "Observation",
        "status": "final",
        "effective": { "dateTime": "2018-05-01T:12:00:00Z" },
        "id": "ldlcholesterol",
        "code": {
          "text": "LDL Chol. (Calc)",
          "coding": [
            {
              "code": "39469-2",
              "system": "http://loinc.org",
              "display": "Cholesterol in LDL [Moles/​volume] in Serum or Plasma by calculation"
            }
          ]
        },
        "subject": { "id": "pat1", "resourceType": "Patient" }
      }
    },
    {
      "resource": {
        "category": [
          {
            "text": "Laboratory",
            "coding": [
              {
                "code": "laboratory",
                "system": "http://hl7.org/fhir/observation-category",
                "display": "Laboratory"
              }
            ]
          }
        ],
        "referenceRange": [
          {
            "high": {
              "code": "mmol/L",
              "unit": "mmol/l",
              "value": 2,
              "system": "http://unitsofmeasure.org"
            }
          }
        ],
        "meta": {
          "lastUpdated": "2018-11-29T20:07:30.603Z",
          "versionId": "336",
          "tag": [{ "system": "https://aidbox.app", "code": "updated" }]
        },
        "value": {
          "Quantity": {
            "code": "mmol/L",
            "unit": "mmol/l",
            "value": 1.3,
            "system": "http://unitsofmeasure.org"
          }
        },
        "resourceType": "Observation",
        "status": "final",
        "effective": { "dateTime": "2018-05-01T:12:00:00Z" },
        "id": "tryglyceride",
        "code": {
          "text": "Tryglyceride",
          "coding": [
            {
              "code": "14927-8",
              "system": "http://loinc.org",
              "display": "Triglyceride [Moles/​volume] in Serum or Plasma"
            }
          ]
        },
        "subject": { "id": "pat1", "resourceType": "Patient" }
      }
    },
    {
      "resource": {
        "code": {
          "coding": [
            {
              "code": "29463-7",
              "system": "http://loinc.org",
              "display": "Body Weight"
            }
          ]
        },
        "value": {
          "Quantity": {
            "code": "[lb_av]",
            "unit": "lbs",
            "value": 181,
            "system": "http://unitsofmeasure.org"
          }
        },
        "status": "final",
        "subject": { "id": "pat1", "resourceType": "Patient" },
        "category": [
          {
            "coding": [
              {
                "code": "vital-signs",
                "system": "http://hl7.org/fhir/observation-category",
                "display": "Vital Signs"
              }
            ]
          }
        ],
        "effective": { "dateTime": "2018-05-05T14:10:00Z" },
        "id": "weight3",
        "resourceType": "Observation",
        "meta": {
          "lastUpdated": "2018-11-29T20:07:30.977Z",
          "versionId": "339",
          "tag": [{ "system": "https://aidbox.app", "code": "updated" }]
        }
      }
    },
    {
      "resource": {
        "code": {
          "text": "Heart rate",
          "coding": [
            {
              "code": "8867-4",
              "system": "http://loinc.org",
              "display": "Heart rate"
            }
          ]
        },
        "value": {
          "Quantity": {
            "code": "/min",
            "unit": "beats/min",
            "value": 74,
            "system": "http://unitsofmeasure.org"
          }
        },
        "status": "final",
        "subject": { "id": "pat1", "resourceType": "Patient" },
        "category": [
          {
            "text": "Vital Signs",
            "coding": [
              {
                "code": "vital-signs",
                "system": "http://hl7.org/fhir/observation-category",
                "display": "Vital Signs"
              }
            ]
          }
        ],
        "effective": { "dateTime": "2018-05-05T10:10:00Z" },
        "id": "heart-rate",
        "resourceType": "Observation",
        "meta": {
          "lastUpdated": "2018-11-29T20:07:29.901Z",
          "versionId": "329",
          "tag": [{ "system": "https://aidbox.app", "code": "updated" }]
        }
      }
    },
    {
      "resource": {
        "category": [
          {
            "text": "Laboratory",
            "coding": [
              {
                "code": "laboratory",
                "system": "http://hl7.org/fhir/observation-category",
                "display": "Laboratory"
              }
            ]
          }
        ],
        "referenceRange": [
          {
            "low": {
              "code": "mmol/L",
              "unit": "mmol/L",
              "value": 1.5,
              "system": "http://unitsofmeasure.org"
            }
          }
        ],
        "meta": {
          "lastUpdated": "2018-11-29T20:07:30.000Z",
          "versionId": "333",
          "tag": [{ "system": "https://aidbox.app", "code": "updated" }]
        },
        "value": {
          "Quantity": {
            "code": "mmol/L",
            "unit": "mmol/L",
            "value": 1.3,
            "system": "http://unitsofmeasure.org"
          }
        },
        "resourceType": "Observation",
        "status": "final",
        "effective": { "dateTime": "2018-05-01T:12:00:00Z" },
        "id": "hdlcholesterol",
        "code": {
          "text": "Cholesterol in HDL",
          "coding": [
            {
              "code": "2085-9",
              "system": "http://loinc.org",
              "display": "Cholesterol in HDL [Moles/​volume] in Serum or Plasma"
            }
          ]
        },
        "subject": { "id": "pat1", "resourceType": "Patient" }
      }
    },
    {
      "resource": {
        "code": {
          "text": "Body height",
          "coding": [
            {
              "code": "8302-2",
              "system": "http://loinc.org",
              "display": "Body height"
            }
          ]
        },
        "value": {
          "Quantity": {
            "code": "[in_i]",
            "unit": "in",
            "value": 71,
            "system": "http://unitsofmeasure.org"
          }
        },
        "status": "final",
        "subject": { "id": "pat1", "resourceType": "Patient" },
        "category": [
          {
            "coding": [
              {
                "code": "vital-signs",
                "system": "http://hl7.org/fhir/observation-category",
                "display": "Vital Signs"
              }
            ]
          }
        ],
        "effective": { "dateTime": "2018-05-05T10:10:00Z" },
        "id": "height",
        "resourceType": "Observation",
        "meta": {
          "lastUpdated": "2018-11-29T20:07:29.913Z",
          "versionId": "331",
          "tag": [{ "system": "https://aidbox.app", "code": "updated" }]
        }
      }
    },
    {
      "resource": {
        "category": [
          {
            "coding": [
              {
                "code": "vital-signs",
                "system": "http://hl7.org/fhir/observation-category",
                "display": "Vital Signs"
              }
            ]
          }
        ],
        "meta": {
          "lastUpdated": "2018-11-29T20:07:29.085Z",
          "versionId": "324",
          "tag": [{ "system": "https://aidbox.app", "code": "updated" }]
        },
        "interpretation": {
          "text": "Below low normal",
          "coding": [
            {
              "code": "L",
              "system": "http://hl7.org/fhir/v2/0078",
              "display": "low"
            }
          ]
        },
        "resourceType": "Observation",
        "component": [
          {
            "code": {
              "text": "Systolic blood pressure",
              "coding": [
                {
                  "code": "8480-6",
                  "system": "http://loinc.org",
                  "display": "Systolic blood pressure"
                }
              ]
            },
            "value": {
              "Quantity": {
                "code": "mm[Hg]",
                "unit": "mmHg",
                "value": 107,
                "system": "http://unitsofmeasure.org"
              }
            },
            "interpretation": {
              "text": "Normal",
              "coding": [
                {
                  "code": "N",
                  "system": "http://hl7.org/fhir/v2/0078",
                  "display": "normal"
                }
              ]
            }
          },
          {
            "code": {
              "text": "Diastolic blood pressure",
              "coding": [
                {
                  "code": "8462-4",
                  "system": "http://loinc.org",
                  "display": "Diastolic blood pressure"
                }
              ]
            },
            "value": {
              "Quantity": {
                "code": "mm[Hg]",
                "unit": "mmHg",
                "value": 60,
                "system": "http://unitsofmeasure.org"
              }
            },
            "interpretation": {
              "text": "Below normal",
              "coding": [
                {
                  "code": "L",
                  "system": "http://hl7.org/fhir/v2/0078",
                  "display": "low"
                }
              ]
            }
          }
        ],
        "status": "final",
        "effective": { "dateTime": "2018-05-05T10:10:00Z" },
        "id": "bp",
        "code": {
          "text": "Blood pressure systolic & diastolic",
          "coding": [
            {
              "code": "85354-9",
              "system": "http://loinc.org",
              "display": "Bood pressure panel with all children optional"
            }
          ]
        },
        "bodySite": {
          "coding": [
            {
              "code": "368209003",
              "system": "http://snomed.info/sct",
              "display": "Right arm"
            }
          ]
        },
        "subject": { "id": "pat1", "resourceType": "Patient" }
      }
    },
    {
      "resource": {
        "category": [
          {
            "text": "Laboratory",
            "coding": [
              {
                "code": "laboratory",
                "system": "http://hl7.org/fhir/observation-category",
                "display": "Laboratory"
              }
            ]
          }
        ],
        "referenceRange": [
          {
            "high": {
              "code": "mmol/L",
              "unit": "mmol/l",
              "value": 4.5,
              "system": "http://unitsofmeasure.org"
            }
          }
        ],
        "meta": {
          "lastUpdated": "2018-11-29T20:07:29.921Z",
          "versionId": "332",
          "tag": [{ "system": "https://aidbox.app", "code": "updated" }]
        },
        "value": {
          "Quantity": {
            "code": "mmol/L",
            "unit": "mmol/l",
            "value": 6.3,
            "system": "http://unitsofmeasure.org"
          }
        },
        "resourceType": "Observation",
        "status": "final",
        "effective": { "dateTime": "2018-05-01T:12:00:00Z" },
        "id": "cholesterol",
        "code": {
          "text": "Cholesterol",
          "coding": [
            {
              "code": "14647-2",
              "system": "http://loinc.org",
              "display": "Cholesterol [Moles/​volume] in Serum or Plasma"
            }
          ]
        },
        "subject": { "id": "pat1", "resourceType": "Patient" }
      }
    },
    {
      "resource": {
        "code": {
          "text": "Body temperature",
          "coding": [
            {
              "code": "8310-5",
              "system": "http://loinc.org",
              "display": "Body temperature"
            }
          ]
        },
        "value": {
          "Quantity": {
            "code": "[degF]",
            "unit": "F",
            "value": 100.4,
            "system": "http://unitsofmeasure.org"
          }
        },
        "status": "final",
        "subject": { "id": "pat1", "resourceType": "Patient" },
        "category": [
          {
            "coding": [
              {
                "code": "vital-signs",
                "system": "http://hl7.org/fhir/observation-category",
                "display": "Vital Signs"
              }
            ]
          }
        ],
        "effective": { "dateTime": "2018-05-05T10:10:00Z" },
        "id": "temperature",
        "resourceType": "Observation",
        "meta": {
          "lastUpdated": "2018-11-29T20:07:30.604Z",
          "versionId": "338",
          "tag": [{ "system": "https://aidbox.app", "code": "updated" }]
        }
      }
    },
    {
      "resource": {
        "category": [
          {
            "text": "Laboratory",
            "coding": [
              {
                "code": "laboratory",
                "system": "http://hl7.org/fhir/observation-category",
                "display": "Laboratory"
              }
            ]
          }
        ],
        "referenceRange": [
          {
            "low": {
              "code": "mmol/L",
              "unit": "mmol/l",
              "value": 3.1,
              "system": "http://unitsofmeasure.org"
            }
          },
          {
            "high": {
              "code": "mmol/L",
              "unit": "mmol/l",
              "value": 6.2,
              "system": "http://unitsofmeasure.org"
            }
          }
        ],
        "meta": {
          "lastUpdated": "2018-11-29T20:07:29.912Z",
          "versionId": "330",
          "tag": [{ "system": "https://aidbox.app", "code": "updated" }]
        },
        "interpretation": {
          "text": "High",
          "coding": [
            {
              "code": "H",
              "system": "http://hl7.org/fhir/v2/0078",
              "display": "High"
            }
          ]
        },
        "value": {
          "Quantity": {
            "code": "mmol/L",
            "unit": "mmol/l",
            "value": 6.3,
            "system": "http://unitsofmeasure.org"
          }
        },
        "resourceType": "Observation",
        "status": "final",
        "effective": { "dateTime": "2018-05-01T:12:00:00Z" },
        "id": "blood-glucose",
        "code": {
          "text": "Glucose [Moles/volume] in Blood",
          "coding": [
            {
              "code": "15074-8",
              "system": "http://loinc.org",
              "display": "Glucose [Moles/volume] in Blood"
            }
          ]
        },
        "subject": { "id": "pat1", "resourceType": "Patient" }
      }
    }
  ]
}
```

**Comment by orlandow** [comment]
It works with devbox:

```
┌────
│ GET /Observation/$lastn?_count=200&_totalMethod=count&subject=pat1&category=vital-signs&max=1
└────

┌────
│ weight3
│ height
│ temperature
│ bp
│ heart-rate
└────
```

But fails to filter in the cloud (same version as devbox)

```
┌────
│ GET /Observation/$lastn?_count=200&_totalMethod=count&subject=pat1&category=vital-signs&max=1
└────

┌────
│ ldlcholesterol
│ tryglyceride
│ weight3
│ heart-rate
│ hdlcholesterol
│ height
│ bp
│ cholesterol
│ temperature
│ blood-glucose
└────
```

**Comment by orlandow** [comment]
Version: `0.3.3-135-g9a0ee34`

**Comment by mlapshin** [comment]
fixed in `0.3.3-140-g9b7aea5`
